packages/valory/skills/abstract_round_abci/tests/test_base_rounds.py

```diff
@@ -393,35 +431,48 @@ def test_vote_count(self) -> None:
 
         self._test_payload_with_wrong_round_count(test_round)
 
-    def test_negative_threshold(
-        self,
-    ) -> None:
-        """Runs test."""
+    @pytest.mark.parametrize("vote", [True, False, None])
+    def test_threshold(self, vote: Optional[bool]) -> None:
+        """Runs threshold test."""
 
         test_round = self.setup_test_voting_round()
-        first_payload, *payloads = get_dummy_tx_payloads(self.participants, vote=False)
+        test_round.collection_key = "dummy_collection_key"
+        test_round.done_event = "DONE_EVENT"
+        test_round.negative_event = "NEGATIVE_EVENT"
+        test_round.none_event = "NONE_EVENT"
+
+        expected_threshold = {
+            True: lambda: test_round.positive_vote_threshold_reached,
+            False: lambda: test_round.negative_vote_threshold_reached,
+            None: lambda: test_round.none_vote_threshold_reached,
+        }[vote]
+
+        expected_event = {
+            True: test_round.done_event,
+            False: test_round.negative_event,
+            None: test_round.none_event,
+        }[vote]
```

Comment:
Why not in the parametrization? E.g.:

```python
@pytest.mark.parametrize("vote, expected_threshold, expected_event", [[True, "positive_vote_threshold_reached", "done_event"], [False...], [None...]])
```

and then use `getattr`.

Reply:
input arguments have a perfect correlation is why I didn't do it